Gems/PhysX/Code/Editor/KinematicDescriptionDialog.cpp

```diff
@@ -0,0 +1,136 @@
+            connect(m_ui->kinematicBox, &QGroupBox::clicked, this, &KinematicDescriptionDialog::OnButtonClicked);
+
+            if (m_kinematicSetting)
+            {
+                m_ui->kinematicRadioButton->setChecked(true);
+            }
+            else
+            {
+                m_ui->dynamicRadioButton->setChecked(true);
+            }
+
+            // running OnButtonClicked manually to get initial border highlight
+            OnButtonClicked();
+        }
+
+        void KinematicDescriptionDialog::OnButtonClicked()
+        {
+            const QString boxStyleSheet = QString("background-color: rgb(51, 51, 51);");
+            const QString boxBorderStyleSheet = QString(" border: 1px solid rgb(30, 112, 235);");
+
+            if (m_ui->dynamicRadioButton->isChecked())
+            {
+                m_ui->dynamicBox->setStyleSheet(boxStyleSheet + boxBorderStyleSheet);
+                m_ui->kinematicBox->setStyleSheet(boxStyleSheet + " border: none;");
+                m_kinematicSetting = false;
+            }
+            else
+            {
+                m_ui->kinematicBox->setStyleSheet(boxStyleSheet + boxBorderStyleSheet);
+                m_ui->dynamicBox->setStyleSheet(boxStyleSheet + " border: none;");
+                m_kinematicSetting = true;
+            }
+
+            UpdateDialogText();
+        }
+
+
+        void KinematicDescriptionDialog::UpdateDialogText()
+        {
+            if (m_kinematicSetting)
+            {
+                m_ui->kinematicRadioButton->setChecked(true);
+
+                m_ui->selectedDescriptionLabel->setText(KinematicDescription);
+
+                m_ui->validLabel1->setText(MoveWithCodeText);
+                m_ui->validLabel2->setText(MoveManuallyText);
+                m_ui->validIcon1->setText(MoveWithCodeIcon);
+                m_ui->validIcon2->setText(MoveManuallyIcon);
+
+                m_ui->invalidLabel1->setText(CollisionsText);
+                m_ui->invalidLabel2->setText(GravityText);
+                m_ui->invalidIcon1->setText(ImpactIcon);
+                m_ui->invalidIcon2->setText(GravityIcon);
+            }
+            else
+            {
+                m_ui->dynamicRadioButton->setChecked(true);
+
+                m_ui->selectedDescriptionLabel->setText(DynamicDescription);
+
+                m_ui->validLabel1->setText(CollisionsText);
+                m_ui->validLabel2->setText(GravityText);
+                m_ui->validIcon1->setText(ImpactIcon);
+                m_ui->validIcon2->setText(GravityIcon);
+
+                m_ui->invalidLabel1->setText(MoveWithCodeText);
+                m_ui->invalidLabel2->setText(MoveManuallyText);
+                m_ui->invalidIcon1->setText(MoveWithCodeIcon);
+                m_ui->invalidIcon2->setText(MoveManuallyIcon);
+            }
+        }
+
+        bool KinematicDescriptionDialog::GetResult() const
+        {
+            return m_kinematicSetting;
+        }
+    }
+}
```

Comment:
uber nit: Closing namespaces would be great here too (clang-format will add these automatically by the way which is great for new files)